### PR TITLE
Details to motivation

### PR DESCRIPTION
**Pull request by hemanth**
Added more info specifically on what we are trying to solve with this proposal

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,54 @@
...
+* We have an `optionsBag`and we would allow on `['shell', 'env', 'extendEnv', 'uid', 'gid']` on it.
+* From a `req.body` we want to extract `['name', 'company', 'email', 'password']`
+* Checking if a component `shouldReload` by extracting `compareKeys` from `props` and compare it with `prevProps`. 
+* Say we have a `depsObject` and we need to ignore all `@internal/packages` from it.
+* We have `props` from which we need to remove `[‘_csrf’, ‘_method’]`
+* We need to construct a `newModelData` by removing `action.deleted` from `({ ...state.models, ...action.update })`
+
+Well, you see life is all about `pick`ing what we want and `omit`ing what we don't! 
+
+Would life be easier if the language provided a convinent method to help us during similar scenarios?
```
Comment (ljharb) [inline README.md:21]:
```suggestion
Would life be easier if the language provided a convenient method to help us during similar scenarios?
```

File: README.md [hunk 2]
```diff
@@ -1,12 +1,54 @@
...
+```js
+const omit = keys => Object.fromEntries(
+    keys.map(k => !obj.hasOwnProperty(k) && [k, obj[k]]).filter(x => x)
+);
+```
+
+The major challenges we see with the above implementations:
+
+* It is not ergonomic 
+* We can't omit for unknown values
```
Comment (ljharb) [inline README.md:45]: this applies to destructuring, but not to your custom function, which take an array

File: README.md [hunk 3]
```diff
@@ -1,12 +1,54 @@
...
+* We need to construct a `newModelData` by removing `action.deleted` from `({ ...state.models, ...action.update })`
+
+Well, you see life is all about `pick`ing what we want and `omit`ing what we don't! 
+
+Would life be easier if the language provided a convinent method to help us during similar scenarios?
+
+Now, one might argue saying we can implement `pick` and `omit` as below:
+
+```js
+const pick = keys => Object.fromEntries(
```
Comment (ljharb) [inline README.md:26]:
```suggestion
const pick = (obj, keys) => Object.fromEntries(
```

File: README.md [hunk 4]
```diff
@@ -1,12 +1,54 @@
...
+);
+
+/*
+We can also use a Destructuring assignment
+const { authKey, ...toLog } = userInfo;
+*/
+```
+
+```js
+const omit = keys => Object.fromEntries(
```
Comment (ljharb) [inline README.md:37]:
```suggestion
const omit = (obj, keys) => Object.fromEntries(
```

File: README.md [hunk 5]
```diff
@@ -1,12 +1,54 @@
...
+    keys.map(k => !obj.hasOwnProperty(k) && [k, obj[k]]).filter(x => x)
+);
+```
+
+The major challenges we see with the above implementations:
+
+* It is not ergonomic 
+* We can't omit for unknown values
+
+We can read more about such use-cases and challenges in [es.discourse](https://es.discourse.group/t/object-prototype-pick-object-prototype-omit/515).
```
Comment (ljharb) [inline README.md:47]: there's like a dozen threads in the discourse and es-discuss; i'd link to them all

File: README.md [hunk 6]
```diff
@@ -1,12 +1,54 @@
...
+
+With that in mind would it not be easier if we had `Object.pick` and `Object.omit` static methods?!
+
+Let us now discuss about what the sytanx of such useful method would be?
```
Comment (ljharb) [inline README.md:51]:
```suggestion
Let us now discuss about what the API of such a useful method would be?
```

File: README.md (hunk 2 above)
Comment (hemanth) [reply README.md:45]: Yes, destructuring is mentioned in the comments section in the above implementation

File: README.md [hunk 7]
```diff
@@ -42,9 +42,15 @@ const omit = (obj, keys) => Object.fromEntries(
 The major challenges we see with the above implementations:
 
 * It is not ergonomic 
-* We can't omit for unknown values
+* We can't omit for unknown values (with destructuring)
```
Comment (ljharb) [inline README.md:45]: the thing to convey tho is that this only works for "omit" with static values. destructuring doesn't work at all for "pick", or for "omit" with dynamic values.

Comment (hemanth) [reply README.md]: nods, rephrased the text.

File: README.md [hunk 8]
```diff
@@ -1,12 +1,61 @@
...
-
-ECMAScript Proposal, specs, and reference implementation for `Object.pick`, `Object.omit`.
+> ECMAScript Proposal, specs, and reference implementation for `Object.pick`, `Object.omit`.
 
 Spec drafted by [@Aleen](https://github.com/aleen42).
 
 ### Motivation
 
-To operate an object convenient by picking or omitting its properties, described in [the group](https://es.discourse.group/t/object-prototype-pick-object-prototype-omit/515).
+Let us consider few scenarios from the real would to understand what we are trying to solve in this propsoal.
```
Comment (aleen42) [inline README.md:8]:
```suggestion
Let us consider a few scenarios from the real world to understand what we are trying to solve in this proposal. 
```

File: README.md [hunk 9]
```diff
@@ -1,12 +1,61 @@
...
+```js
+const omit = (obj, keys) => Object.fromEntries(
+    keys.map(k => !obj.hasOwnProperty(k) && [k, obj[k]]).filter(x => x)
+);
+```
+
+The major challenges we see with the above implementations:
+
+* It is not ergonomic!
+* If we opt the destructuring way it doesn't work at all for `pick`, or for `omit` with dynamic values.
```
Comment (aleen42) [inline README.md:46]:
```suggestion
* If we opt for the destructuring way it doesn't work at all for `pick`, or for `omit` with dynamic values.
```

File: README.md [hunk 10]
```diff
@@ -1,12 +1,61 @@
...
+
+* [Object.{pick,omit}](https://es.discourse.group/t/object-prototype-pick-object-prototype-omit/515).
+* [Object restructuring syntax](https://es.discourse.group/t/object-restructuring-syntax/651)
+* [Object Array Pick](https://es.discourse.group/t/object-array-pick/992)
+* [js-pick-notation](https://github.com/rtm/js-pick-notation)
+* [slect multiple object values](https://es.discourse.group/t/set-multiple-object-values-a-b-undefined/1052/3)
+
+With that in mind would it not be easier if we had `Object.pick` and `Object.omit` static methods?!
+
+Let us now discuss about what the API of such a useful method would be?
```
Comment (aleen42) [inline README.md:58]:
```suggestion
Let us now discuss what the API of such a helpful method would be?
```

File: README.md (hunk 7 above)
Comment (aleen42) [reply README.md]: Like something:

1. Destructuring cannot clone a new object while `Object.pick` can
2. Destructuring cannot pick up properties from the prototype while `Object.pick` can
3. Destructuring cannot pick properties dynamically, while `Object.pick` can
4. Destructuring cannot omit some properties, and we can only clone and delete without this proposal
5. ...

**Comment by hemanth** [comment]
This is ready enough now, will be merging this PR, thanks for the reviews.

**Pull request merged**